bagel/pheno_utils.py

```diff
@@ -191,6 +192,55 @@ def are_inputs_compatible(data_dict: dict, pheno_df: pd.DataFrame) -> bool:
     return all([key in pheno_df.columns for key in data_dict.keys()])
 
 
+def find_undefined_categorical_column_values(
+    data_dict: dict, pheno_df: pd.DataFrame
+) -> dict:
+    """
+    Checks that all categorical column values have annotations. Returns a dictionary containing
+    any categorical column names and specific column values not defined in the corresponding data
+    dictionary entry.
+    """
+    all_undefined_values = {}
+    for col, attr in data_dict.items():
+        if is_column_categorical(col, data_dict):
+            known_values = list(attr["Levels"].keys()) + attr[
+                "Annotations"
+            ].get("MissingValues", [])
+            # NOTE: (also applies to find_unused_missing_values) The below comparison block could also be
```

Comment:
Not sure this comment is needed. I don't think anyone is going to read this section in the future and go and scratch their head why you didn't implement it differently. I think I'd remove it